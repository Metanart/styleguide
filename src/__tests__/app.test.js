import React from 'react';
import { create, act } from 'react-test-renderer';
import App from '../app';
import AppProps from './app-view.mockup.json';

describe('App generic tests', () => {
    let root;

    it('When app is open', () => {
        act(() => {
            root = create(<App sections={JSON.parse(AppProps.openApp).sections} />);
        });

        expect(root.toJSON()).toMatchSnapshot();
    });
});
