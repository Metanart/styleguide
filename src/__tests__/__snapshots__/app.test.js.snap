// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App generic tests When app is open 1`] = `
<div
  className="sc-fznzOf paqyy"
>
  <aside
    className="sc-AxjAm cRLJtR"
  >
    <div
      className="sc-AxirZ iluSoF"
    >
      <div
        className="sc-AxiKw bcuorQ"
      >
        <div
          className="sc-AxhCb jVvKPm"
        >
          <svg
            viewBox="0 0 44 44"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="#7000E3"
              height="44"
              rx="3.9"
              width="44"
            />
            <path
              d="M28 11.3c-2.3 0-4.6 1-6 3-1.4-2-3.7-3-6-3-4.3 0-7.8 3.3-7.8 7.6 0 6.5 9.3 15 13.8 15s13.8-8.5 13.8-15c0-4.4-3.5-7.7-7.8-7.7"
              fill="#FF671B"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-AxgMl otYGI"
      >
        <input
          autoComplete="off"
          className="sc-AxheI fleRyR"
          onChange={[Function]}
          placeholder="Search component name"
          type="text"
          value=""
        />
        <div
          className="sc-Axmtr crOZCD"
        >
          <div
            className="sc-AxhUy hmNFUn"
            role="presentation"
            size={0}
          >
            <svg
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.5 20.92l7.62 7.67-1.43 1.4-7.72-7.79A10.83 10.83 0 0 1 2 13.28v-.45a10.84 10.84 0 0 1 21.67 0v.45c0 2.98-1.21 5.68-3.16 7.64zM4 12.82v.46a8.83 8.83 0 0 0 17.67 0v-.45a8.83 8.83 0 0 0-17.67 0z"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
      <nav
        className="sc-fznyAO bkLnjh"
      >
        <div
          className="sc-fznKkj jkDQak"
        >
          <div
            className="sc-AxmLO kzOOLP"
          >
            <div
              className="sc-fzozJi ejeJSx"
              onClick={[Function]}
            >
              Structure
            </div>
            <div
              className="sc-fzoLsD iUeVUF"
            >
              <div
                className="sc-fzpans cIwqpl"
              >
                <a
                  className="sc-fzplWN hXlTqQ"
                  data-vrt-locator="link"
                  href="#Structure-Component"
                >
                  Component
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-fznKkj jkDQak"
        >
          <div
            className="sc-AxmLO kzOOLP"
          >
            <div
              className="sc-fzozJi ejeJSx"
              onClick={[Function]}
            >
              Components
            </div>
            <div
              className="sc-fzoLsD iUeVUF"
            >
              <div
                className="sc-fzpans cIwqpl"
              >
                <a
                  className="sc-fzplWN hXlTqQ"
                  data-vrt-locator="link"
                  href="#Components-ErrorBoundary"
                >
                  ErrorBoundary
                </a>
              </div>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </aside>
  <main
    className="sc-fznZeY iASHnM"
  >
    <article
      className="sc-fzoXWK khMgHS"
      id="welcome!"
    >
      <header
        className="styleguide-component__header"
      >
        <h1
          className="sc-fzpmMD cahimd"
        >
          Welcome!
        </h1>
        <div
          className="sc-fznxKY hrnXgF"
        >
          Style guide is a tool to illustrate, sandbox and test your components.
        </div>
      </header>
    </article>
  </main>
</div>
`;
